xbmc/guilib/GUIPrimitive.cpp

```diff
@@ -0,0 +1,113 @@
+/*
+ *  Copyright (C) 2024 Team Kodi
+ *  This file is part of Kodi - https://kodi.tv
+ *
+ *  SPDX-License-Identifier: GPL-2.0-or-later
+ *  See LICENSES/README.md for more information.
+ */
+
+#include "GUIPrimitive.h"
+
+#include "Interpolators.h"
+
+#include <cassert>
+
+using namespace KODI::GUILIB;
+
+CGUIPrimitive::CGUIPrimitive(
+    int parentID, int controlID, float posX, float posY, float width, float height)
+  : CGUIImage(parentID, controlID, posX, posY, width, height, CTextureInfo())
+{
+  ControlType = GUICONTROL_PRIMITIVE;
+}
+
+void CGUIPrimitive::SetUniform(GUIINFO::CGUIInfoColor& color)
```

Comment:
```suggestion
void CGUIPrimitive::SetUniform(GUIINFO::CGUIInfoColor color)
```
I suggest to pass by value because of the small size, if not as const&.